worker/worker.go

```diff
@@ -0,0 +1,54 @@
+package worker
+
+import (
+	"encoding/json"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"log"
+	"os"
+)
+
+type PBWorker struct {
+	ServerEndPoint string
+	Vcpu           string
+	Ram            string
+	Image          string
+	Address        string
+	Token          string
+	Id             string
+	QueueId        string
+}
+
+func (w *PBWorker) Subscribe() {
+	httpResponse := utils.Post(w.Id, &PBWorker{Ram: w.Ram, Vcpu: w.Vcpu, Image: w.Image,
+		Address: w.Address, Id: w.Id, QueueId: w.QueueId}, w.ServerEndPoint + "/workers")
+
+	if httpResponse.StatusCode != 201 {
+		log.Fatal("The work could not be subscribed")
+	}
+
+	var parsedBody map[string]string
+	json.Unmarshal(httpResponse.Body, &parsedBody)
+
+	w.Token = parsedBody["arrebol-worker-token"]
```

Comment:
When you access a map it also returns a boolean which indicates whether the key is present in the map or not, it is super useful to be aware of any error that might be spread through the code execution.

`value, ok := map[key]`